src/makeInfinityAdder.js

```diff
@@ -6,7 +6,26 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let summary = 0;
+
+  function adder(x) {
+    if (x !== undefined) {
+      return (y) => {
+        if (y !== undefined) {
+          summary += x + y;
+
+          return adder(x + y);
+        }
+        summary = 0;
+
+        return x;
+      };
+    }
```

Comment:
```suggestion
    if (x !== undefined) {
      summary += x;

      return adder;
    }
```

Reply:
So easy that way XD? I have been implementing this task for 5 hours straight through two inner functions XD